### ISSUE TITLE
Weird error

### ISSUE THREAD

**Issue by astrogames06** [bevyengine/bevy#4611]
## Bevy version

0.7.0

## Operating system

WSL ubuntu

## What i did

I ran ```cargo run```

## What you expected to happen

Window creation

## What actually happened

error: failed to run custom build command for `libudev-sys v0.1.4`

Caused by:
  process didn't exit successfully: `/mnt/d/rust_raylib_test/target/debug/build/libudev-sys-96ea0176be42e1de/build-script-build` (exit status: 101)   
  --- stdout
  cargo:rerun-if-env-changed=LIBUDEV_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=LIBUDEV_STATIC
  cargo:rerun-if-env-changed=LIBUDEV_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu      
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu      
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu    
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu    
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "`\"pkg-config\" \"--libs\" \"--cflags\" \"libudev\"` did not exit successfully: exit status: 1\nerror: could not find system library 'libudev' required by the 'libudev-sys' crate\n\n--- stderr\nPackage libudev was not found in the pkg-config search path.\nPerhaps you should add the directory containing `libudev.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'libudev' found\n"', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/libudev-sys-0.1.4/build.rs:38:41
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed

## main.rs

``` rust
use bevy::prelude::*;

fn main() {
    App::build()
        .insert_resource(ClearColor(Color::rgb(0.04, 0.04, 0.04)))
        .insert_resource(WindowDescriptor {
            title: "Rust Invaders!".to_string(),
            width: 598.0,
            heght: 676.0,
            ..Default::default()
        })
        .add_plugins(DefaultPlugins)
        .run();
}
```

## Cargo.toml

``` rust
[package]
name = "rust_raylib_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
bevy = "0.7.0"

```

**Comment by mockersf** [comment]
Did you install the dependencies? forUbuntu that would be https://github.com/bevyengine/bevy/blob/main/docs/linux_dependencies.md#ubuntu-2004

**Comment by MilkMouseBearCoder** [comment]
> Did you install the dependencies? forUbuntu that would be https://github.com/bevyengine/bevy/blob/main/docs/linux_dependencies.md#ubuntu-2004

Now when I run ```cargo run```

## I get this error

ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4528:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4528:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4528:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5007:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM default
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: NoDevice', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_audio-0.5.0/src/audio_output.rs:22:67
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by devil-ira** [comment]
> thread 'main' panicked at 'called Result::unwrap() on an Err value: NoDevice', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_audio-0.5.0/src/audio_output.rs:22:67

You are using an old version of Bevy. You should update to version `0.7`.
Bevy `0.5` panics if it can't find an audio device.

**Comment by MilkMouseBearCoder** [comment]
> > thread 'main' panicked at 'called Result::unwrap() on an Err value: NoDevice', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_audio-0.5.0/src/audio_output.rs:22:67
> 
> You are using an old version of Bevy. You should update to version `0.7`. Bevy `0.5` panics if it can't find an audio device.

How do I make a simple window in bevy 0.7

**Comment by MilkMouseBearCoder** [comment]
> > thread 'main' panicked at 'called Result::unwrap() on an Err value: NoDevice', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_audio-0.5.0/src/audio_output.rs:22:67
> 
> You are using an old version of Bevy. You should update to version `0.7`. Bevy `0.5` panics if it can't find an audio device.

I updated it but now when I run ```cargo run``` I get this error

thread 'main' panicked at 'Failed to initialize any backend! Wayland status: "backend disabled" X11 status: LibraryOpenError(OpenError { kind: Library, detail: "opening library failed (libXcursor.so.1: cannot open shared object file: No such file or directory); opening library failed (libXcursor.so: cannot open shared object file: No such file or directory)" })', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/linux/mod.rs:619:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

## Code

``` rust
use bevy::prelude::*;

fn main() {
    App::new()
        .insert_resource(ClearColor(Color::rgb(0.04, 0.04, 0.04)))
        .insert_resource(WindowDescriptor {
            title: "Bevy".to_string(),
            width: 598.0,
            height: 676.0,
            ..Default::default()
        })
        .add_plugins(DefaultPlugins)
        .run();
}
```

**Comment by ArikRahman** [comment]
Same issue for me on Pop!_OS

**Comment by SolarLiner** [comment]
Fixed by #2269, it should not panic when not being able to find a suitable audio device anymore.

Also a duplicate of #316 #850 #2661 and a bunch of others
